docs/templates.rst

```diff
@@ -3,4 +3,102 @@
 Templates
 =========
 
-WRITE ME
+This document describes how to use templates with Django Bread.
+
+Template Context
+----------------
+
+In addition to the context variables that Django Vanilla Views provides, Django Bread provides the
+following to all of your templates::
+
+        Variable             Description
+        --------             -----------
+        bread                Bread object
+        verbose_name         Model's verbose_name field
+        verbose_name_plural  Model's verbose_name_plural field
+        base_template        The default base_template that should be extended.
+        may_{action}         (i.e. may_browse, may_read, etc.) Boolean describing whether user has specified permission
+
+Certain action views provide some additional variables. The Browse view has ``columns`` and
+``filter``. The Read view has ``form``.
+
+Override ``get_context_data`` to change the variables provided to your template:
+
+.. code-block:: python
+
+   class MyBrowseView(BrowseView):
+       def get_context_data(self, **kwargs):
+           data = super(BrowseView, self).get_context_data(**kwargs)
+           data['my_special_var'] = 42
+           return data
+
+
+Template Resolution
+-------------------
+
+There are various ways to configure how and in what order Django Bread searches for the proper
+template for your view.
+
+By default, each Django Bread view will search first in an app-specific location (described below),
+and then fall back to the templates provided by the Django Bread package.
+
+App-specific location
+^^^^^^^^^^^^^^^^^^^^^
+
+The app-specific location is described by this pattern: ``{app_label}/{model}_{view}.html``.
+For example, if your app's name is MyApp and your model's name is MyModel, Django Bread will look
+for a browse template in ``myapp/mymodel_browse.html``.
+
+Customization
+^^^^^^^^^^^^^
+
+There are 3 ways that you can customize this behavior even further:
+
+1. Specify ``template_name_pattern`` in your Bread object. All views in that Bread object will then
+   search in the app-specific location first (described above), then search
+   ``template_name_pattern``, and then fall back to the Django Bread provided templates. The value
+   supplied to ``template_name_pattern`` is a string that can take zero or more placeholders with
+   the names ``app_label``, ``model``, or ``view``. An example of a valid pattern would be
+   ``'{app_label}/special_{model}_{view}.html'``.
+
+2. Create a subclass of Bread and set ``template_name_pattern`` in that subclass. Then, use that
```

Comment:
I guess this isn't really another way to do it, it's just taking advantage of Python to do number 1 more efficiently, right?  Maybe this should just be a parenthetical suggestion on the end of number 1.  Or maybe I'm being too picky :-)